### ISSUE TITLE
[Doc]: Add alt text to figures in documentation 

### ISSUE THREAD

**Issue by story645** [matplotlib/matplotlib#24309]
### Problem

None of our images have alt-text and that's becoming a standard now and our [communication's guidelines](https://github.com/matplotlib/governance/blob/main/communications_guidelines.md#accessibility) say we should add alt-text when the platform allows.

### Suggested improvement

The steps involved in having a consistent way to add alt text to our docs are

#### code
- https://github.com/matplotlib/matplotlib/pull/21328
- [x] #29649 
- [ ] enable support of alt text in [sphinx gallery scraper](https://github.com/sphinx-gallery/sphinx-gallery/blob/e2ecb35d9049d0df0a1db4f6d15be817745cd565/sphinx_gallery/scrapers.py#L89)
  -  https://github.com/sphinx-gallery/sphinx-gallery/blob/e2ecb35d9049d0df0a1db4f6d15be817745cd565/sphinx_gallery/scrapers.py#L437-L448
- [ ] add passing through alt-text to any file types that support it
     - [ ]  svg is included in #21328 
    - [ ]  pdf https://github.com/matplotlib/matplotlib/issues/24309#issuecomment-1297192256

#### docs
- [ ] replace [alt text guidelines](https://matplotlib.org/governance/communications_guidelines.html#accessibility) in communication guidelines with a standalone documentation page on alt-text for visualization best practices, update links in communications and pull requests guidelines to post to it. 
- [ ] write tutorial on making accessible viz
- [ ] document using alt_aria as part of and alt text and update the [PR template](https://github.com/matplotlib/matplotlib/blob/65c1b89f96f9aff58fd6462429fcd8ff93d3ba51/.github/PULL_REQUEST_TEMPLATE.md) w/ a nudge to use alt text
- [ ] add alt text to gallery files: I do it here in order of visibility, but traffic is probably a better way to prioritize)
   - [ ] plot gallery - it feeds the home page gallery so is probably highest profile
   - [ ] tutorials - doesn't always have code for the images
   - [ ] gallery 
   - [ ] narrative text
   - [ ] older whats news and anything missing from the above
  
Adding alt text is piles of work and probably would benefit from strategy and partnerships w/ classes/sprints or hiring someone depending on timeline.

**Comment by oscargus** [comment]
PDF does also support alt-texts (and has some other accessibility support). See 14.9.3 Alternate Descriptions in https://opensource.adobe.com/dc-acrobat-sdk-docs/pdfstandards/PDF32000_2008.pdf

**Comment by oscargus** [comment]
I just sent an email to a Swedish researcher asking for some pointers. Let us see if I get some answer. Also, found this paper which seems quite relevant: https://content.iospress.com/articles/technology-and-disability/tad170181 (I have not read it yet, but I guess the references and citing papers can be worthwhile, even if we may not be interested in automation in the first place).

**Comment by tacaswell** [comment]
https://dev.exiv2.org/projects/exiv2/wiki/The_Metadata_in_PNG_files

https://iptc.org/std/photometadata/specification/IPTC-PhotoMetadata#alt-text-accessibility

https://exiv2.org/tags.html / https://www.cipa.jp/std/documents/e/DC-008-2012_E.pdf

There are reasonable places to put "alt text" (although IPTC makes a distinction between "ald text" and "image description").

----

options:

 - minimal: nothing (but docs)! currently `fig.savefig('...png', metadata={'alt': 'The alt text'})` works (any key works), also works `fig.savefig('/tmp/test.svg', metadata={'Description': 'This is alt text'})` , `fig.savefig('/tmp/test.pdf', metadata={'Alt': 'This is alt text'})` also works (but warns).  Other raster formats do not work (fails in `imsave`), but that is a matter of sorting out the right invocation to pillow.
 - medium: add a `alt_text` keyword to `savefig` that is threaded through each of the backends to translate to the correct name for that file format
 - medium++: add a `alt_text` or `description` attribute to `Figure` that is the default value threaded through `savefig(..., alt_text)`

open questions:
 - do any screen readers _actually_ look at these?
 - if there are conflicting standards, how to we pick with standard to follow?

Other tasks:
 - if we do medium++, teach matplotlib-inline to extract the same for the html
 - if we do the medium++, teach sphinx-gallery to extract the same
